### ISSUE TITLE
Override the Pages layout?

### ISSUE THREAD

**Issue by kberridge** [statiqdev/Statiq#614]
I'm new to wyam, I've been reading the docs but I'm stumped.  Is there a way to override the layout used by the Blog recipe to render pages?  I see it's using _Layout and I would like to create a _PageLayout similar to _PostLayout for my theme.

**Comment by kberridge** [comment]
To be more clear, the layout I want to change is currently set by recipes\Wyam.Blog\Blog.cs on line 182 where RenderPages is initialized and the layout is set to "/_Layout.cshtml".  I would like to somehow set that to "/_PageLayout.cshtml".

**Comment by daveaglick** [comment]
Sure thing - the way theme and layout files are selected is by cascading through different paths until the requested file is found. First the path inside the theme NuGet package is searched, then a folder called `theme` (if it exists), and then `input`. If a file is found in a latter folder, that version is used instead of the former folder. 

For example, if the build looks for `/_Layout.cshtml` and finds it in the theme package, that’ll be used. If however it’s _also_ in the `input` folder, that file will be used instead.

TL;DR: to replace the main layout for a given theme, add a `_Layout.cshtml` to your `input` folder. You could also go deeper and change the layout file name, but that would require replacing a pipeline or writing a custom config instead of using the recipe.

**Comment by kberridge** [comment]
Hi Dave thanks for the reply!

If I put a _Layout.cshtml file in the root of the input folder so that it applies to "pages" it also applies to "posts", actually it applies to all pages.  I want to customize the layout of pages only, for example the about page.

Specifically my posts have a different padding than my index and archive pages, and I want the pages to have the same padding as posts.  But since pages only use _Layout I can't customize them in the same way I did with _PostLayout, that's why I was hoping to create a _PageLayout.

**Comment by daveaglick** [comment]
Ah, got it. Unfortunately that's not available out of the box for any of the official themes. It's still doable though by replacing that specific pipeline with your own (in `config.wyam`):

```
int renderPagesIndex = Pipelines.IndexOf(Blog.RenderPages);
Pipelines.Remove(Blog.RenderPages);
Pipelines.Insert(
  renderPagesIndex,
  Blog.RenderPages,
  (IPipeline)new Wyam.Web.Pipelines.RenderPages(
    Blog.RenderPages,
    new Wyam.Web.Pipelines.RenderPagesSettings
    {
      Pipelines = new string[] { Blog.Pages },
      Layout = (doc, ctx) => "/_PageLayout.cshtml"
    }));
```

Having to get the pipeline index, remove the old one, and then insert the new one in it's place is definitely a bit goofy - there's issue #591 to add a `.Replace()` method for just this kind of thing. Otherwise, we're basically copying the code from the recipe for that pipeline into your config file as a replacement pipeline with a tweak to the layout file. The only downside is that this may make your config a little brittle - if the code for this pipeline changes in the recipe you'll need to manually keep up with that.

**Comment by kberridge** [comment]
Just what I needed, thanks so much!
